[GEN-234] Created a settings view with delete and sign out buttons

abhipi opened the pull request:
- Created a Settings View with the Delete and Sign Out Buttons, and their associated methods (handleDeleteAccount and the imported SignOut).
- Removed the redundant components on the Home View and Profile View (Delete Button changed to a Settings Wheel).
- Passed the navigation prop, through UserProfile to Climber/Setter Profile  for navigation to Settings.
- Imported AuthContext in Settings.js to handle account deletion and sign outs.

VGmt commented:
@abhipi  – When the 'Files Changed' count is exceptionally high, please include a brief explanation in your PR comment. This will help alleviate the initial shock of seeing '404 files changed' 😅.

Otherwise, your comments look good. The review will be conducted tomorrow.

VGmt commented:
Additionally, including an image of the implemented feature helps _**tremendously**_

abhipi commented:
> Additionally, including an image of the implemented feature helps _**tremendously**_

Ofc 😆 i just added a new package 'react-native-vector-icons/MaterialIcons'. This caused the node_module changes. Attaching images of the features below: 

The Settings View (Identical for both):
<img width="263" alt="image" src="https://github.com/nagimonyc/trk-app/assets/75244191/3cce5a63-ca66-4c80-b770-f6e196d06b24">

For the Climber:
<img width="260" alt="image" src="https://github.com/nagimonyc/trk-app/assets/75244191/efd41fe3-e891-4ef7-b307-83c1983a770b">

For the Setter:
<img width="264" alt="image" src="https://github.com/nagimonyc/trk-app/assets/75244191/f754283a-4a35-4b6b-ad27-195d994cc6ea">

Sorry for the confusion!

abhipi commented:
> > Additionally, including an image of the implemented feature helps _**tremendously**_
> 
> Ofc 😆 i just added a new package 'react-native-vector-icons/MaterialIcons'. This caused the node_module changes. Attaching images of the features below:
> 
> The Settings View (Identical for both): <img alt="image" width="263" src="https://user-images.githubusercontent.com/75244191/279546227-3cce5a63-ca66-4c80-b770-f6e196d06b24.png">
> 
> For the Climber: <img alt="image" width="260" src="https://user-images.githubusercontent.com/75244191/279545975-efd41fe3-e891-4ef7-b307-83c1983a770b.png">
> 
> For the Setter: <img alt="image" width="264" src="https://user-images.githubusercontent.com/75244191/279546379-f754283a-4a35-4b6b-ad27-195d994cc6ea.png">
> 
> Sorry for the confusion!

Corrected It! Do let me know if this looks fine.

VGmt commented:
Much better 👍

VGmt commented on trk/src/Navigation/AppNavigator.js:
```diff
@@ -35,6 +36,7 @@ function ProfileStack() {
       <Stack.Screen name="Profile" component={UserProfile} />
       <Stack.Screen name="Detail" component={ClimbDetailScreen} />
       <Stack.Screen name="Set" component={SetDetail} />
+      <Stack.Screen name="Your Settings" component={Settings} />
```
Naming conventions would have the Stack Screen be called: 'Settings', instead of 'Your Settings'.

### Strava
![0E3C694E-2A1D-4F15-AC95-D451904F5744_1_102_o](https://github.com/nagimonyc/trk-app/assets/66965648/b1a30005-35c6-4d50-9d9b-91be727dcab7)

### ChatGPT
![E5EAB1B3-F15D-4EE1-8019-F1EFF4381ECB_1_102_o](https://github.com/nagimonyc/trk-app/assets/66965648/fa1a1493-2afb-4195-9d21-805e145b6682)

### NFC Tools
![8BFA37A6-2062-47C8-988D-4D83343AB2CB_4_5005_c](https://github.com/nagimonyc/trk-app/assets/66965648/2ed22baa-6ddc-45a5-9ac1-1eac527cfcd5)


⚠️ Changing this name field will have repercussions on your 'navigation.navigate'.

VGmt commented on trk/src/Components/ClimberProfile.js:
```diff
@@ -31,41 +32,15 @@ const ClimberProfile = () => {
                 <View style={styles.header}>
                     <Text style={styles.titleText}>Activity</Text>
-                    <Button title="Delete Account" onPress={handleDeleteAccount}></Button>
+                    <TouchableOpacity onPress={() => navigation.navigate('Your Settings')}>
```
^ here

VGmt commented on trk/src/Components/SetterProfile.js:
```diff
@@ -29,42 +27,15 @@ const SetterProfile = (props) => {
                 <View style={styles.header}>
                     <Text style={styles.titleText}>Activity</Text>
-                    <Button title="Delete Account" onPress={handleDeleteAccount}></Button>
+                    <TouchableOpacity onPress={() => navigation.navigate('Your Settings')}>
```
^ here

abhipi replied:
@VGmt Made the changes!

VGmt replied:
Perfect. Thanks!

VGmt reviewed (changes requested):
Cf. changes in Stack Screen name
